src/preset_cli/api/clients/superset.py

```diff
@@ -477,7 +477,17 @@ def get_database(self, database_id: int) -> Any:
         """
         Return a single database.
         """
-        return self.get_resource("database", database_id)
+        database = self.get_resource("database", database_id)
+        if "sqlalchemy_uri" in database:
+            return database
+
+        url = self.baseurl / "api/v1/database" / str(database_id) / "connection"
+        response = self.session.get(url)
+        validate_response(response)
+
+        resource = response.json()["result"]
```

Comment:
it will be nice to have a schema for the return object

Reply:
I am not familiar with the above 2 endpoints but another option is to introduce a new method for the connection endpoint.
The API client should be as dumb as possible.

Reply:
Good point, I have a TODO in that file to add them. It's not super trivial because the schema depends on the version of Superset running.